streams/src/test/java/org/apache/kafka/streams/StreamsBuilderTest.java

```diff
@@ -403,4 +416,69 @@ public void shouldThrowExceptionWhenTopicNamesAreNull() {
         builder.stream(Arrays.asList(null, null));
         builder.build();
     }
+
+    @Test
+    public void shouldUseSpecifiedNameForStreamSourceProcessor() {
+        final String expected = "source-processor";
+        builder.stream(STREAM_TOPIC, Consumed.as(expected));
+        builder.stream(STREAM_TOPIC_TWO);
+        builder.build();
+        final ProcessorTopology topology = builder.internalTopologyBuilder.rewriteTopology(new StreamsConfig(props)).build();
+        assertSpecifiedNameForOperation(topology, expected, "KSTREAM-SOURCE-0000000000");
+    }
+
+    @Test
+    public void shouldUseSpecifiedNameForTableSourceProcessor() {
+        final String expected = "source-processor";
+        builder.table(STREAM_TOPIC, Consumed.as(expected));
+        builder.table(STREAM_TOPIC_TWO);
+        builder.build();
+        final ProcessorTopology topology = builder.internalTopologyBuilder.rewriteTopology(new StreamsConfig(props)).build();
+
+        assertSpecifiedNameForOperation(topology,
+                expected,
+                expected + "-table-source",
+                "KSTREAM-SOURCE-0000000002",
```

Comment:
Why is this not zero but two? Seems we "burn" two indexes, but I am not sure why atm? Also, why is it `KSTREAM` instead of `KTABLE` ?

Reply:
First, for topology backward-compatibility we still increment the counter used for naming processor. Then, builder.table() actually generates two nodes, one source node and one table node. Finally, we have a state store created for the table. The state store name is generated first this why if we don't provide a custome source name the node will be indexed -00001. However, the state store is not visible within the topology description. I guess this is because the state is considered not queriable.